com.unity.render-pipelines.universal/Runtime/RendererFeatures/ScreenSpaceAmbientOcclusion.cs

```diff
@@ -270,6 +331,18 @@ public override void Execute(ScriptableRenderContext context, ref RenderingData
                     // Set the global SSAO texture and AO Params
                     cmd.SetGlobalTexture(k_SSAOTextureName, m_SSAOTexture2Target);
                     cmd.SetGlobalVector(k_SSAOAmbientOcclusionParamName, new Vector4(0f, 0f, 0f, m_CurrentSettings.DirectLightingStrength));
+
+                    // If true, SSAO pass is inserted after opaque pass and is expected to modulate lighting result now.
+                    if (m_CurrentSettings.AfterOpaque)
+                    {
+                        // This implicitely also bind depth attachment. Explicitely binding m_Renderer.cameraDepthTarget does not work.
+                        cmd.SetRenderTarget(
+                            m_Renderer.cameraColorTarget,
+                            RenderBufferLoadAction.Load,
+                            RenderBufferStoreAction.Store
+                        );
+                        cmd.DrawMesh(RenderingUtils.fullscreenMesh, Matrix4x4.identity, material, 0, (int)ShaderPasses.AfterOpaque);
```

Comment:
Will this work in XR? @DennisDeRykeUnity @kaychang-unity please check XR path was tested with AfterOpaque

Reply:
I believe @DennisDeRykeUnity tested it already.

The api `cmd.SetRenderTarget()` is not great when dealing with a RenderTargetIdentifier that point to combined color+depth surfarce. I don't think the behaviour is well defined.

Reply:
@phi-lira You asked me to "check XR path was tested with AfterOpaque" but I do not know what this means.  I tested Quest in Link mode (basically make the Quest act like a Rift, tethered to my PC via USB) using test scenes 142 and 144 (modified slightly to enable XRSDK+Oculus and pose driver for camera tracking).  Would I need to skim through the shader in those scenes and look for an AfterOpaque pass?

Reply:
@DennisDeRykeUnity Ah I got it! "AfterOpaque" is a mode for SSAO where it is applied after opaque geometry lighting instead of before (the default). If you check the SSAO renderer feature UI, there is a checkbox "After Opaque" that will move SSAO application after opaque. The AO will be slightly different and "darker", but it is expected (because less physically correct).